source/src_pw/hamilt_hip.cpp

```diff
@@ -0,0 +1,678 @@
+    }
+
+	if (nb == 1 || nb >= nstart)
+    {
+        lwork = 2 * nstart; // mohan modify 2009-08-02
+    }
+    else
+    {
+        lwork = (nb + 1) * nstart;
+    }
+
+    std::complex<double> *work = new std::complex<double>[lwork];
+	ModuleBase::GlobalFunc::ZEROS(work, lwork);
+
+    //=====================================================================
+    // input s and (see below) h are copied so that they are not destroyed
+    //=====================================================================
+
+    int info = 0;
+    int rwork_dim;
+    if (all_eigenvalues)
+    {
+        rwork_dim = 3*nstart-2;
+    }
+    else
+    {
+        rwork_dim = 7*nstart;
+    }
+
+    double *rwork = new double[rwork_dim];
+    ModuleBase::GlobalFunc::ZEROS( rwork, rwork_dim );
+
+    if (all_eigenvalues)
+    {
+        //===========================
+        // calculate all eigenvalues
+        //===========================
+        hvec = hc;
+        LapackConnector::zhegv(1, 'V', 'U', nstart, hvec , ldh, sdum, ldh, e, work , lwork , rwork, info);
+    }
+    else
+    {
+        //=====================================
+        // calculate only m lowest eigenvalues
+        //=====================================
+        int *iwork = new int [5*nstart];
+        int *ifail = new int[nstart];
+
+        ModuleBase::GlobalFunc::ZEROS(rwork,7*nstart);
+        ModuleBase::GlobalFunc::ZEROS(iwork,5*nstart);
+        ModuleBase::GlobalFunc::ZEROS(ifail,nstart);
+
+        hdum.create(nstart, ldh);
+        hdum = hc;
+
+    	//=============================
+    	// Number of calculated bands
+    	//=============================
+    	int mm = nbands;
+
+        LapackConnector::zhegvx
+        (
+                1,      //INTEGER
+                'V',    //CHARACTER*1
+                'I',    //CHARACTER*1
+                'U',    //CHARACTER*1
+                nstart, //INTEGER
+                hdum,   //COMPLEX*16 array
+                ldh,    //INTEGER
+                sdum,   //COMPLEX*16 array
+                ldh,    //INTEGER
+           		0.0,    //DOUBLE PRECISION
+                0.0,    //DOUBLE PRECISION
+                1,      //INTEGER
+                nbands, //INTEGER
+                0.0,    //DOUBLE PRECISION
+                mm,     //INTEGER
+                e,      //DOUBLE PRECISION array
+                hvec,   //COMPLEX*16 array
+                ldh,    //INTEGER
+                work,   //DOUBLE array, dimension (MAX(1,LWORK))
+                lwork,  //INTEGER
+                rwork , //DOUBLE PRECISION array, dimension (7*N)
+                iwork,  //INTEGER array, dimension (5*N)
+                ifail,  //INTEGER array, dimension (N)
+                info    //INTEGER
+        );
+
+        delete[] iwork;
+        delete[] ifail;
+    }
+    delete[] rwork;
+    delete[] work;
+
+	ModuleBase::timer::tick("Hamilt","diagH_LAPACK");
+    return;
+}
+
+/*
+void Hamilt::diagH_CUSOLVER(
```

Comment:
What's this function for?

Reply:
This function is not used in the ROCm version，it and has been deleted.